oc-aliases.sh

```diff
@@ -2,75 +2,82 @@
+	selector=$1
+	oc -n ${PROJECT_NAME} logs --selector=${selector} --all-containers
 }
+
 function podbash {
-	podlabel=$1
-	oc exec -it $(oc get pods -n spectrum-discover -l app=${podlabel} -o=jsonpath='{.items[0].metadata.name}') -n=spectrum-discover -- /bin/bash
+	selector=$1
+	deployment=$(oc -n ${PROJECT_NAME} get deployment --selector=${selector} -o name)
+	oc -n ${PROJECT_NAME} rsh ${deployment}
 }
+
 function delpod {
-	podlabel=$1
-	oc delete pod $(oc get pods -n spectrum-discover -l app=${podlabel} -o=jsonpath='{.items[0].metadata.name}') -n=spectrum-discover
+	selector=$1
+	oc -n ${PROJECT_NAME} delete pod --selector=${selector}
 }
+
 function redeploy {
 	deployment=$1
-	newtag=$2
-	oc patch deployment.v1.apps/spectrum-discover-${deployment} -n spectrum-discover -p "{\"spec\":{\"template\":{\"spec\":{\"containers\":[{\"name\":\"spectrum-discover\",\"image\":\"$(kubectl get deployment/spectrum-discover-${deployment} -n spectrum-discover -o=jsonpath='{@.spec.template.spec.containers[0].image}' | sed -e 's/[0-9]*-[0-9]*$//')${newtag}\"}]}}}}"
+	image=$2
+	container="${3}:-spectrum-discover"
+	oc -n ${PROJECT_NAME} patch deployment/${deployment} -p "{\"spec\":{\"template\":{\"spec\":{\"containers\":[{\"name\":\"${container}\",\"image\":\"${image}\"}]}}}}"
 }
 
 # UI Frontend log
-alias uiflog='podlog spectrum-discover-ui-frontend'
+alias uiflog='podlog role=ui,sub-role=frontend'
 # UI Backend log
-alias uiblog='podlog spectrum-discover-ui-backend'
+alias uiblog='podlog role=ui,sub-role=backend'
 # Shell into the running db2wh-rest container
-alias db2ex='podbash spectrum-discover-db2whrest'
+alias db2ex='podbash role=db2whrest'
 # Shell into the running policy engine container
-alias peex='podbash spectrum-discover-policyengine'
+alias peex='podbash role=policyengine'
 # Shell into the running connection management container
-alias conex='podbash spectrum-discover-connmgr'
+alias conex='podbash role=connmgr'
 # Delete db2wh-rest pod
-alias redb2='delpod spectrum-discover-db2whrest'
+alias redb2='delpod role=db2whrest'
 # Db2wh-rest log
-alias db2log='podlog spectrum-discover-db2whrest'
+alias db2log='podlog role=db2whrest'
 # Connection management log
-alias conlog='podlog spectrum-discover-connmgr'
+alias conlog='podlog role=connmgr'
 # Policy Engine log
-alias pollog='podlog spectrum-discover-policyengine'
-alias pelog='podlog spectrum-discover-policyengine'
+alias pollog='podlog role=policyengine'
+alias pelog='podlog role=policyengine'
 # SD Monitor log
-alias sdmonlog='podlog spectrum-discover-sdmonitor'
+alias sdmonlog='podlog role=sdmonitor'
 # Db2wh password
-alias pw='oc exec -it -n=spectrum-discover $(oc get pods -n=spectrum-discover -l app=spectrum-discover-connmgr -o=jsonpath={.items[0].metadata.name}) env |grep DB2WHREST_PASSWORD | cut -d "=" -f 2'
+alias pw='oc -n ${PROJECT_NAME} exec -c spectrum-discover $(oc -n ${PROJECT_NAME} get pods -l role=connmgr -o name) -- env | grep DB2WHREST_PASSWORD | cut -d "=" -f 2'
```

Comment:
This pw command helps the user easily retrieve the bluadmin password. When we deploy Discover, the source of this password is the LDAP secret, as noticed recently by the architect. So, instead doing this complex command, we should just retrieve that password from the secret with the following:
`alias pw='oc get secret/c-isd-ldappassword -n ${PROJECT_NAME} -o jsonpath="{.data.password}"'`

Reply:
This approach does not work for self managed db2uclusters, maybe we can implement something to look for the ldap password but that requires additional logic in code, since changing the approach is low priority we can defer this improvement until next update